src/cpp/src/sequence_group.hpp

```diff
@@ -396,7 +421,17 @@ class SequenceGroup {
     }
 
     size_t get_num_logical_blocks() const {
-        return (get_context_len() + m_block_size - 1) / m_block_size;
+        size_t max_num_logical_blocks = 0;
+        for (const auto& seq : m_sequences) {
+            max_num_logical_blocks = std::max(max_num_logical_blocks,
+                                              (get_context_len() - seq->get_num_evicted_tokens() + m_block_size - 1) / m_block_size);
+        }
+        return max_num_logical_blocks;
```

Comment:
each running sequence is supposed to have the same length within a group 
so, do we need a loop here and even touch this method?

IMO, when we have performed eviction, context_len should be updated to lower value and eviction information should not be stored within sequence group.

Reply:
Refactored, should not longer be a problem